starter-web-servlet/build.gradle

```diff
@@ -15,11 +15,11 @@ dependencies {
     implementation project(":starter-api")
     implementation "io.micronaut:micronaut-inject"
 
-    implementation("io.micronaut.servlet:micronaut-servlet-engine:1.0.0.M3")
```

Comment:
is this no longer necessary?

Reply:
It doesn't seem to be...  It's pulled in by https://github.com/micronaut-projects/micronaut-starter/blob/a32c415b5671fc58a71cd27f6c62feeced4d1b65/starter-web-servlet/build.gradle#L18

But now I see that's `developmentOnly`, not `implementation` 🤔 

I'll try building the war and deploying it to jetty locally...

What is `starter-web-servlet` actually for?  Do we deploy a war anywhere?  Is it used?

Reply:
Ok, so we need to have

```
    implementation "io.micronaut.servlet:micronaut-http-server-jetty"
```

Otherwise when we try to deploy the WAR to Jetty with:

```
docker \
    run \
    -p 8080:8080 -v $(pwd)/starter-web-servlet/build/libs/starter-web-servlet-3.3.0-SNAPSHOT.war:/var/lib/jetty/webapps/ROOT.war jetty:9.4.44-jre8
```

we get

```
2022-01-28 14:02:14.046:WARN:oejw.WebAppContext:main: Failed startup of context o.e.j.w.WebAppContext@71318ec4{ROOT,/,file:///tmp/jetty/jetty-0_0_0_0-8080-ROOT_war-_-any-2049695182268382873/webapp/,UNAVAILABLE}{/var/lib/jetty/webapps/ROOT.war}
java.lang.NoSuchMethodError: io.micronaut.context.ApplicationContext.build()Lio/micronaut/context/ApplicationContextBuilder;
	at io.micronaut.servlet.engine.initializer.MicronautServletInitializer.buildApplicationContext(MicronautServletInitializer.java:57)
	at io.micronaut.servlet.engine.initializer.MicronautServletInitializer.onStartup(MicronautServletInitializer.java:35)
	at org.eclipse.jetty.plus.annotation.ContainerInitializer.callStartup(ContainerInitializer.java:142)
	at org.eclipse.jetty.annotations.ServletContainerInitializersStarter.doStart(ServletContainerInitializersStarter.java:64)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:356)
	at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1449)
	at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1414)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:910)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
	at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.eclipse.jetty.deploy.bindings.StandardStarter.processBinding(StandardStarter.java:46)
```

Reply:
TL;DR:

You were right, I added it back in...

Resolving this